tensorflow_addons/metrics/f1_scores.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements F1 scores."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.keras.metrics import Metric
+import numpy as np
+
+
+class F1_micro(Metric):
+    """
+    Calculates F1 micro score
+    """
+    def __init__(self, name='f1-score'):
+        super(F1_micro, self).__init__(name=name)
+        self.true_positives = self.add_weight('TP-class', shape=[],
```

Comment:
Nit: TP-class -> TP_class

Reply:
A suggestion here: maybe "true_positives" is a better name to match core TF.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics.py#L1160

Reply:
Yup, it's better we match core TF.